### PR TITLE
Allows field-dependent SimpleForcing. Updates plankton growth example.

### PR DESCRIPTION
**Pull request by glwagner**
A 'field-dependent' forcing is a user-specified forcing term that depends on `x, y, z, t, u`, where `u` is the field being forced, and optionally parameters.

### CODE REVIEW

File: examples/ocean_convection_with_plankton.jl [hunk 1]
```diff
@@ -50,10 +50,10 @@ nothing # hide
 # ## Define a forcing function
 #
 # Our forcing function roughly corresponds to the growth of phytoplankton in light
-# (with a penetration depth of 16 meters here), and death due to natural mortality
+# (with a penetration depth of 16 meters here), and death due to grazing
 # at a rate of 1 phytoplankton unit per second.
 
-growth_and_decay = SimpleForcing((x, y, z, t) -> exp(z/16))
+growth_and_decay = SimpleForcing((x, y, z, t) -> exp(z/16) - 1)
```
Comment (ali-ramadhan) [inline examples/ocean_convection_with_plankton.jl:56]: Nice fix :microbe:

File: src/Forcing/simple_forcing.jl [hunk 2]
```diff
@@ -1,52 +1,142 @@
 """
-    SimpleForcing{X, Y, Z, F, P}
+    struct SimpleForcing{X, Y, Z, M, F, P}
```
Comment (ali-ramadhan) [inline src/Forcing/simple_forcing.jl:2]: Unbelievably minor but I think Julia convention is to not write `struct` in the docstring.

For example:
```
help?> Complex

  Complex{T<:Real} <: Number

  Complex number type with real and imaginary part of type T.
```

Comment (glwagner) [reply src/Forcing/simple_forcing.jl]: done!

File: examples/ocean_convection_with_plankton.jl (hunk 1 above)
Comment (glwagner) [reply examples/ocean_convection_with_plankton.jl:56]: these phytoplankton are dying quite fast.

**Pull request merged**